posts/indu-plp-part2/plp-part2.qmd

```diff
@@ -0,0 +1,281 @@
+    WHERE cohort_definition_id = 2
+"""
+diabetes_df = execute(conn, diabetes_query) |> DataFrame
+```
+
+Next, we perform a **left join** with the previously created `features_df`, which contains covariates for all patients in the hypertension cohort. This ensures every patient has their features preserved, and only those present in the diabetes cohort will have a $1$ under the outcome column.
+
+For patients not found in the outcome cohort, the join results in a missing value. We treat these as negative cases $0$, meaning the patient did not develop diabetes during the follow-up period.
+
+```julia
+df = leftjoin(features_df, diabetes_df, on=:subject_id)
+df[!, :outcome] .= coalesce.(df[!, :outcome], 0)
+```
+This gives us a binary classification dataset with:
+
+- $1$: patient developed diabetes after hypertension
+- $0$: patient did not develop diabetes (or not within the observed period)
+
+This labeled dataset is now ready for preprocessing and model training in later steps.
+
+<br>
+<center>
+  ![](./binary_classification.png)
+
+  Binary Classification Model
+</center>
+
+# Step 3: Preprocessing for Modeling
+
+Once we have the raw feature matrix and outcome labels, the next crucial step is to preprocess the dataset to make it suitable for training machine learning models. This step is essential because raw healthcare data, even when structured via the OMOP CDM, often contains missing values, variable scales, and unencoded categorical information. These issues can affect model performance, stability, and interpretability.
+
+The preprocessing approach we use is directly inspired by best practices discussed in the JAMIA paper. The authors highlight the importance of handling missing values, standardizing predictors, and transforming categorical data into usable formats.
+
+We perform the following three tasks:
+
+- Handling missing values
+- Standardizing numeric features
+- Encoding categorical variables
+
+**File:** `preprocessing.jl`
+
+```julia
+# Impute missing values
+for col in names(df)
+    if eltype(df[!, col]) <: Union{Missing, Number}
+        df[!, col] = coalesce.(df[!, col], 0)
+    else
+        df[!, col] = coalesce.(df[!, col], "unknown")
+    end
+end
+
+# Standardize numerical columns
+for col in [:age, :condition_count, :drug_count]
+    μ, σ = mean(skipmissing(df[!, col])), std(skipmissing(df[!, col]))
+    df[!, col] .= (df[!, col] .- μ) ./ σ
+end
+
+# Encode categoricals
+df.gender_concept_id = categorical(df.gender_concept_id)
+df.race_concept_id = categorical(df.race_concept_id)
+```
+
+**Train-Test Splitting**
+
+Finally, we split the data into training and testing sets using an 80-20 split. This ensures we can evaluate how well the model generalizes to unseen data.
+
+```julia
+using MLJ
+train, test = partition(eachindex(df.outcome), 0.8, shuffle=true)
+```
+<br>
+<center>
+  ![](./train_test_splitting.png)
+
+  Train-Test splitting (80% - 20%)
+</center>
+
+# Step 4: Model Training with MLJ.jl
```

Comment:
I really like the code snippets but could you make this section more concise? Thanks!